EventListener/UserContextSubscriber.php

```diff
@@ -87,11 +92,14 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
+        $hash = $this->hashGenerator->generateHash();
+
         if (!$this->requestMatcher->matches($event->getRequest())) {
+            $this->hash = $hash;
+
             return;
         }
 
-        $hash = $this->hashGenerator->generateHash();
```

Comment:
hm, good point. i think this idea makes sense. we don't want to trigger a session for anonymous here. and we don't use the hash in the sanity check below if there was no hash header.